pkg/hintrunner/hinter/scope.go

```diff
@@ -65,36 +86,48 @@ func (sm *ScopeManager) AssignVariables(values map[string]any) error {
 	return nil
 }
 
-func (sm *ScopeManager) GetVariableValue(name string) (any, error) {
+func (sm *ScopeManager) GetScopeValue(name string) (*ScopeValue, error) {
 	scope, err := sm.getCurrentScope()
 	if err != nil {
 		return nil, err
 	}
 
 	if value, ok := (*scope)[name]; ok {
-		return value, nil
+		return &value, nil
```

Comment:
Returning pointer to ScopeValue here might cause dangling pointer. Would be better to return the value itself.